### PR TITLE
Updates to Predicate Design and Addition of Bitcoin Scopes

### PR DESCRIPTION
**Pull request by natebuch**
## Description

Changes:
Update predicate design documentation.  
- There was repeated information between the Stacks and Bitcoin predicate page.  Combined the documentation into a single page for better continuity.
- Updated predicate-design file
- Deleted bitcoin predicates file
- Deleted stacks predicates file

Add Bitcoin scopes reference file.
- Created reference page for Bitcoin scopes
- Nested Stacks and Bitcoin references pages under a scopes file

Reference Links
- Updated existing guides to point to new predicate design page.
- Updated existing guides to point to Bitcoin scopes, Stacks scopes or both based on context.

### CODE REVIEW

File: content/docs/stacks/chainhook/concepts/predicate-design.mdx [hunk 1]
```diff
@@ -1,64 +1,346 @@
 ---
-title: Predicate design
+title: Predicate Design
```
Comment (ryanwaits) [inline content/docs/stacks/chainhook/concepts/predicate-design.mdx:2]:
```suggestion
title: Predicate design
```

File: content/docs/stacks/chainhook/concepts/predicate-design.mdx [hunk 2]
```diff
@@ -1,64 +1,346 @@
...
 description: Predicates are the building blocks of Chainhook.
+toc: true
```
Comment (ryanwaits) [inline content/docs/stacks/chainhook/concepts/predicate-design.mdx:4]:
```suggestion
```

**Comment by ryanwaits** [comment]
@natebuch appreciating the contributions here, this looks great! 

I'm assuming you removed the `stacks-predicates` and `bitcoin-predicates` files in favor of the new sections under the API References?

If so, we'll need to make sure we handle redirects for those pages. You can take a look at the `vercel.json` file where we handles those.

Redirects needed:
-  `/stacks/chainhook/concepts/stacks-predicates`
- `/stacks/chainhook/concepts/bitcoin-predicates`
- `/stacks/chainhook/references/scopes`

Example:

```json
{
   "source": "/stacks/chainhook/concepts/bitcoin-predicates",
   "destination": "/stacks/chainhook/references/bitcoin-scopes",
   "permanent": true
},
```

**Comment by natebuch** [comment]
Redirects have been updated.

**Pull request merged**